src/CoreWCF.Http/tests/DependencyInjection/ServiceTypeErrorMessageTests.cs

```diff
@@ -0,0 +1,122 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using CoreWCF.Configuration;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Helpers;
+using Xunit.Abstractions;
+
+namespace CoreWCF.Http.Tests.DependencyInjection
+{
+    public class ServiceTypeErrorMessageTests
+    {
+        private readonly ITestOutputHelper _output;
+
+        public ServiceTypeErrorMessageTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact]
+        public void ServiceWithoutDefaultConstructorAndNotInDI_ShowsServiceTypeInErrorMessage()
+        {
+            // Arrange - Create a host with a service that has no default constructor and is not registered in DI
+            IWebHost host = ServiceHelper.CreateWebHostBuilder<StartupWithProblematicService>(_output).Build();
+            
+            using (host)
+            {
+                // Act - Start the host (this should succeed)
+                host.Start();
+                
+                // Create a client to make a request that will trigger service instantiation
+                System.ServiceModel.BasicHttpBinding httpBinding = ClientHelper.GetBufferedModeBinding();
+                var factory = new System.ServiceModel.ChannelFactory<ITestService>(httpBinding,
+                    new System.ServiceModel.EndpointAddress(new Uri($"http://localhost:{host.GetHttpPort()}/testservice")));
+                ITestService channel = factory.CreateChannel();
+                
+                // Assert - Making a request should throw an exception with the service type name
+                var exception = Assert.ThrowsAny<Exception>(() => channel.GetData("test"));
+                
+                // Output detailed exception information for debugging
+                _output.WriteLine($"Exception type: {exception.GetType().FullName}");
+                _output.WriteLine($"Exception message: {exception.Message}");
+                if (exception.InnerException != null)
```

Comment:
The InnerException is unimportant, there won't be one. Remove anything about the inner exception